guides/source/working_with_javascript_in_rails.md

```diff
@@ -3,561 +3,197 @@
+    end
+  end
+end
+```
 
-Certain elements of your page aren't actually referring to any URL, but you may want
-them to trigger Ajax calls. Specifying the `data-url` attribute along with
-the `data-remote` one will trigger an Ajax call to the given URL. You can also
-specify extra parameters through the `data-params` attribute.
+Rails will automatically look for a `.turbo_stream.erb` view file and render that view when found.
 
-This can be useful to trigger an action on check-boxes for instance:
+Turbo Stream responses can also be rendered inline in the controller action:
 
-```html
-<input type="checkbox" data-remote="true"
-    data-url="/update" data-params="id=10" data-method="put">
+```ruby
+def create
+  @post = Post.new(post_params)
+
+  respond_to do |format|
+    if @post.save
+      format.turbo_stream { render turbo_stream: turbo_stream.prepend('posts', partial: 'post') }
+    else
+      format.html { render :new, status: :unprocessable_entity }
+    end
+  end
+end
 ```
 
-#### `data-type`
-
-It is also possible to define the Ajax `dataType` explicitly while performing
-requests for `data-remote` elements, by way of the `data-type` attribute.
+Finally, Turbo Streams can be initiated from a model or a background job using built-in helpers. These broadcasts can be used to update content via a WebSocket connection to all users, keeping page content fresh and bringing your application to life.
 
-### Confirmations
+To broadcast a Turbo Stream from a model combine a model callback like this:
 
-You can ask for an extra confirmation of the user by adding a `data-confirm`
-attribute on links and forms. The user will be presented with a JavaScript `confirm()`
-dialog containing the attribute's text. If the user chooses to cancel, the action
-doesn't take place.
+```ruby
+class Post < ApplicationRecord
+  after_create_commit { broadcast_append_to('posts') }
+end
+```
 
-Adding this attribute on links will trigger the dialog on click, and adding it
-on forms will trigger it on submit. For example:
+With a WebSocket connection set up on the page that should receive the updates like this:
 
 ```erb
-<%= link_to "Dangerous zone", dangerous_zone_path,
-  data: { confirm: 'Are you sure?' } %>
+<%= turbo_stream_from "posts" %>
 ```
 
-This generates:
+Learn more about Turbo Streams in the [Turbo Handbook](https://turbo.hotwired.dev/handbook/streams).
 
-```html
-<a href="..." data-confirm="Are you sure?">Dangerous zone</a>
-```
+Replacements for Rails/UJS functionality
+————————————
 
-The attribute is also allowed on form submit buttons. This allows you to customize
-the warning message depending on the button which was activated. In this case,
-you should **not** have `data-confirm` on the form itself.
+Rails UJS allowed developers to override the `method` of `<a>` tags to perform non-GET requests after a hyperlink click and to add confirmation dialogs before executing an action.
 
+Turbo provides built in replacements for both.
```

Comment:
Since UJs is soft deprecated, I'm not sure it should be mentioned here, as it would confuse new users.

Reply:
I left this in since its a pretty common question for folks migrating to Turbo from Rails UJS. Perhaps they'll just figure it out but acknowledging the transition feels correct to me.

Reply:
Okay, how about this then

> Rails ships with a tool called UJs that allows developers to override the `method` of `<a>` tags to perform non-GET requests after a hyperlink click and to add confirmation dialogs before executing an action. This was the default before Rails 7, but it is now recommended to use Turbo instead.

Reply:
Made a small edit here since Rails no longer ships with UJS.